src/CMakeLists.txt

```diff
@@ -63,7 +63,9 @@ if(NOT CMAKE_BUILD_TYPE)
           FORCE
   )
 endif()
-if(NOT DEFINED CMAKE_INTERPROCEDURAL_OPTIMIZATION AND CMAKE_BUILD_TYPE MATCHES "Rel")
+if(NOT DEFINED CMAKE_INTERPROCEDURAL_OPTIMIZATION
+   AND (CMAKE_BUILD_TYPE MATCHES "Release" OR CMAKE_BUILD_TYPE MATCHES "MinSizeRel")
+)
```

Comment:
Wait what. So this enables LTO when `CMAKE_INTERPROCEDURAL_OPTIMIZATION` is not passed on the command line? That's again the wrong default.

I'm not using `setup.py` personally because it does things I don't want, and I'm building directly using CMake and Ninja -- and in that case it would implicitly enable LTO for me.

I'd say, remove this whole `if`?

Reply:
I was just trying to mimic how PyBInd1sets LTO and do the same for all the subprojects so if it is set, it's propogated down to the subprojects properly.

Reply:
@mosra Enabling LTO is also the default Release config for the emscriptem toolchain though. It would enable in full release mode.